racket/src/cs/rumble/intmap.ss

```diff
@@ -390,9 +384,12 @@
           (let ([xs (Co-pairs a)])
             (and (fx= (length xs) (length (Co-pairs b)))
                  (let loop ([xs xs])
-                   (cond [(null? xs) #t]
-                         [($collision-has-key? et b (caar xs)) (loop (cdr xs))]
-                         [else #f])))))]
+                   (cond [(null? xs)
+                          #t]
+                         [else
+                          (let ([pair ($collision-ref et b (caar xs) values #f)])
+                            (and pair
+                                 (eql? (cdar xs) (cdr pair))))])))))]
```

Comment:
Well, this isn't right either. In my haste, I forgot to recur... Patch forthcoming.

Reply:
Updated, with a better test.